x/chainlink/keeper/querier_test.go

```diff
@@ -0,0 +1,267 @@
+// Copyright 2021 ChainSafe Systems
+// SPDX-License-Identifier: MIT
+
+package keeper
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/ChainSafe/chainlink-cosmos/x/chainlink/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/testutil/testdata"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+func TestQuerier_GetRoundFeedData(t *testing.T) {
+	keeper, ctx := setupKeeper(t)
+	amino := codec.NewLegacyAmino()
+	querier := NewQuerier(*keeper, amino)
+	roundStore := ctx.KVStore(keeper.roundStoreKey)
+
+	testCases := []struct {
+		feedId    string
+		roundId   uint64
+		feedData  []byte
+		submitter []byte
+		insert    bool
+	}{
+		{feedId: "feed1", roundId: 100, feedData: []byte{'a', 'b', 'c'}, submitter: []byte("addressMock1"), insert: true},
+		{feedId: "feed1", roundId: 200, feedData: []byte{'d', 'e', 'f'}, submitter: []byte("addressMock2"), insert: true},
+		{feedId: "feed1", roundId: 300, feedData: []byte{'g', 'h', 'i'}, submitter: []byte("addressMock3"), insert: false},
+		{feedId: "feed2", roundId: 400, feedData: []byte{'j', 'k', 'l'}, submitter: []byte("addressMock4"), insert: false},
+	}
+
+	// Add all feed cases to store
+	for _, tc := range testCases {
+		if !tc.insert {
+			continue
+		}
+		// force set roundId-1 for SetFeedData
+		roundStore.Set(types.GetRoundIdKey(tc.feedId), i64tob(tc.roundId-1))
+
+		msgFeedData := types.MsgFeedData{
+			FeedId:    tc.feedId,
+			FeedData:  tc.feedData,
+			Submitter: tc.submitter,
```

Comment:
should probably include `Signatures` and `IsFeedDataValid` parameters in the type https://github.com/ChainSafe/chainlink-cosmos/blob/master/x/chainlink/types/tx.pb.go#L723

Reply:
+ test cases